Sources/_Datadog_Private/ObjcAppLaunchHandler.m

```diff
@@ -56,32 +57,60 @@ @interface AppLaunchHandler : NSObject
 @implementation AppLaunchHandler
 
 + (void)load {
+    pthread_rwlock_init(&rwLock, NULL);
+
     // This is called at the `_Datadog_Private` load time, keep the work minimal
     FrameworkLoadTime = CFAbsoluteTimeGetCurrent();
 
     NSNotificationCenter * __weak center = NSNotificationCenter.defaultCenter;
-    id __block token = [center
-                        addObserverForName:UIApplicationDidBecomeActiveNotification
-                        object:nil
-                        queue:NSOperationQueue.mainQueue
-                        usingBlock:^(NSNotification *_){
-
-        pthread_rwlock_init(&rwLock, NULL);
+    id __weak __block token = [center addObserverForName:UIApplicationDidBecomeActiveNotification
+                                                  object:nil
+                                                   queue:NSOperationQueue.mainQueue
+                                              usingBlock:^(NSNotification *_){
         pthread_rwlock_wrlock(&rwLock);
-        TimeToApplicationDidBecomeActive = ComputeProcessTimeFromStart();
+        UIApplicationDidBecomeActiveNotificationTime = CFAbsoluteTimeGetCurrent();
         pthread_rwlock_unlock(&rwLock);
-
         [center removeObserver:token];
-        token = nil;
     }];
 }
 
 @end
 
+@implementation UIApplication (Tracking)
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        method_exchangeImplementations(
+            class_getInstanceMethod(self, @selector(init)),
+            class_getInstanceMethod(self, @selector(dd_init))
+        );
+    });
+}
+
+- (instancetype)dd_init {
+    pthread_rwlock_wrlock(&rwLock);
+    UIApplicationInstantiateSingletonTime = CFAbsoluteTimeGetCurrent();
```

Comment:
How this time is different from "will finish launching" and "did finish launching" in app delegate? Let's say the app has **very long** "`dd_init` to _did become active_" measure, then:
* what can be the root cause?
* how actionable it would be (e.g. how it can be optimised)?
* will "`Datadog.initialize()` to _did become active_" also be very long in that case?